server/src/main/java/org/apache/iotdb/db/query/dataset/NewEngineDataSetWithoutValueFilter.java

```diff
@@ -327,6 +328,167 @@ public TSQueryDataSet fillBuffer(int fetchSize, WatermarkEncoder encoder) throws
 
     return tsQueryDataSet;
   }
+  
+  /**
+   * for RPC in RawData query between client and server
+   * fill time buffer, value buffers and bitmap buffers
+   */
+  public TSQueryNonAlignDataSet fillNonAlignBuffer(int fetchSize, WatermarkEncoder encoder) throws IOException, InterruptedException {
+    int seriesNum = seriesReaderWithoutValueFilterList.size();
+    TSQueryNonAlignDataSet tsQueryNonAlignDataSet = new TSQueryNonAlignDataSet();
+
+    PublicBAOS[] timeBAOSList = new PublicBAOS[seriesNum];
+    PublicBAOS[] valueBAOSList = new PublicBAOS[seriesNum];
+    PublicBAOS[] bitmapBAOSList = new PublicBAOS[seriesNum];
+
+    for (int seriesIndex = 0; seriesIndex < seriesNum; seriesIndex++) {
+      timeBAOSList[seriesIndex] = new PublicBAOS();
+      valueBAOSList[seriesIndex] = new PublicBAOS();
+      bitmapBAOSList[seriesIndex] = new PublicBAOS();
+    }
+
+    // used to record a bitmap for every 8 row record
+    int[] currentBitmapList = new int[seriesNum];
```

Comment:
record -> records
optional: BitMap (and other occurrences)-> nullValueBitMap, so it would be more clear what this bit map is intended for.

Reply:
I have removed the bitmap in non align dataset, since it's useless.